R/RestRserveRequest.R

```diff
@@ -91,12 +91,154 @@ RestRserveRequest = R6::R6Class(
         self$method = method
       }
       self$path_parameters = list()
+      self$context = new.env(parent = emptyenv())
       private$id = uuid::UUIDgenerate(TRUE)
+    },
+    get_header = function(name) {
+      if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
+        checkmate::assert_string(name)
+      }
+      name = tolower(name)
+      return(self$headers[[name]])
+    },
+    set_header = function(name, value) {
+      if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
+        checkmate::assert_string(name)
+        checkmate::assert_string(value)
+      }
+      name = tolower(name)
+      self$headers[[name]] = value
+      return(value)
+    },
+    delete_header = function(name) {
```

Comment:
I think we don't need set/delete/append methods for request objects

Reply:
For example the middleware functions need it.

Reply:
middleware functions might need it but still use case look a bit artificial. Point here is that I don't think this will be used 99% of the time and hence I would prefer to not expose these methods. If you have a good example of a use case i can reconsider my point of view.

Reply:
Removed.